Cesium3DTilesSelection/src/BatchTableToGltfStructuralMetadata.cpp

```diff
@@ -50,21 +236,71 @@ struct BinaryProperty {
-const std::map<std::string, GltfFeatureTableType>
-    batchTableComponentTypeToGltfType = {
-        {"BYTE", GltfFeatureTableType{"INT8", sizeof(int8_t)}},
-        {"UNSIGNED_BYTE", GltfFeatureTableType{"UINT8", sizeof(uint8_t)}},
-        {"SHORT", GltfFeatureTableType{"INT16", sizeof(int16_t)}},
-        {"UNSIGNED_SHORT", GltfFeatureTableType{"UINT16", sizeof(uint16_t)}},
-        {"INT", GltfFeatureTableType{"INT32", sizeof(int32_t)}},
-        {"UNSIGNED_INT", GltfFeatureTableType{"UINT32", sizeof(uint32_t)}},
-        {"FLOAT", GltfFeatureTableType{"FLOAT32", sizeof(float)}},
-        {"DOUBLE", GltfFeatureTableType{"FLOAT64", sizeof(double)}},
+const std::map<std::string, GltfPropertyTableType> batchTableTypeToGltfType = {
+    {"SCALAR",
+     GltfPropertyTableType{
+         ExtensionExtStructuralMetadataClassProperty::Type::SCALAR,
+         1}},
+    {"VEC2",
+     GltfPropertyTableType{
+         ExtensionExtStructuralMetadataClassProperty::Type::VEC2,
+         2}},
+    {"VEC3",
+     GltfPropertyTableType{
+         ExtensionExtStructuralMetadataClassProperty::Type::VEC3,
+         3}},
+    {"VEC4",
+     GltfPropertyTableType{
+         ExtensionExtStructuralMetadataClassProperty::Type::VEC4,
+         4}},
+};
+
+struct GltfPropertyTableComponentType {
+  std::string componentType;
+  size_t componentTypeSize;
+};
+
+const std::map<std::string, GltfPropertyTableComponentType>
+    batchTableComponentTypeToGltfComponentType = {
+        {"BYTE",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::INT8,
+             sizeof(int8_t)}},
+        {"UNSIGNED_BYTE",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::UINT8,
+             sizeof(uint8_t)}},
+        {"SHORT",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::INT16,
+             sizeof(int16_t)}},
+        {"UNSIGNED_SHORT",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::UINT16,
+             sizeof(uint16_t)}},
+        {"INT",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::INT32,
+             sizeof(int32_t)}},
+        {"UNSIGNED_INT",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::UINT32,
+             sizeof(uint32_t)}},
+        {"FLOAT",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::
+                 FLOAT32,
+             sizeof(float)}},
+        {"DOUBLE",
+         GltfPropertyTableComponentType{
+             ExtensionExtStructuralMetadataClassProperty::ComponentType::
+                 FLOAT64,
+             sizeof(double)}},
 };
 
 int64_t roundUp(int64_t num, int64_t multiple) noexcept {
```

Comment:
Technically `EXT_structural_metadata` property tables don't need to be aligned to 8-bytes. Quote:

> Implementation note: Authoring tools may choose to align all buffer views to 8-byte boundaries for consistency, but client implementations should only depend on 8-byte alignment for buffer views containing 64-bit component types.

 Should I keep this rounding code? Or okay to remove?

Reply:
There's still a 4-byte alignment requirement though, right? That might be required of all bufferViews in general.

Reply:
It's a bit more nuanced than that, actually:
https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#data-alignment

Reply:
Ah right, I forgot about that. We can just leave the rounding behavior at 8-bytes for simplicity.